jekyll/_cci2/server/installation/phase-3-execution-environments.adoc

```diff
@@ -0,0 +1,957 @@
+----
++
+[source,shell]
+----
+gcloud compute firewall-rules create "circleci-vm-service-external-fw" --network "<network>" --action allow --rules "TCP:54782"
+----
+
+. *Create user*
++
+We recommend you create a unique service account used exclusively by VM Service. The Compute Instance Admin (Beta) role is broad enough to allow VM Service to operate. If you wish to make permissions more granular, you can use the Compute Instance Admin (beta) role documentation as reference.
++
+[source,shell]
+----
+gcloud iam service-accounts create circleci-server-vm --display-name "circleci-server-vm service account"
+----
++
+NOTE: If your are deploying CircleCI server in a shared VCP, you should create this user in the project in which you intend to run your VM jobs.
+
+. *Get the service account email address*
++
+[source,shell]
+----
+gcloud iam service-accounts list --filter="displayName:circleci-server-vm service account" --format 'value(email)'
+----
+
+. *Apply role to service account*
++
+Apply the Compute Instance Admin (Beta) role to the service account:
++
+[source,shell]
+----
+gcloud projects add-iam-policy-binding <YOUR_PROJECT_ID> --member serviceAccount:<YOUR_SERVICE_ACCOUNT_EMAIL> --role roles/compute.instanceAdmin --condition=None
+----
++
+And
++
+[source,shell]
+----
+gcloud projects add-iam-policy-binding <YOUR_PROJECT_ID> --member serviceAccount:<YOUR_SERVICE_ACCOUNT_EMAIL> --role roles/iam.serviceAccountUser --condition=None
+----
+
+. *Enable Workload Identity for Service Account*
++
+This step is required only if you are using link:https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity[Workload Identities] for GKE. Steps to enable Workload Identities are link:https://circleci.com/docs/2.0/server-3-install-prerequisites/index.html#enabling-workload-identity-in-gke[here]
++
+[source,shell]
+----
+gcloud iam service-accounts add-iam-policy-binding <YOUR_SERVICE_ACCOUNT_EMAIL> \
+    --role roles/iam.workloadIdentityUser \
+    --member "serviceAccount:<GCP_PROJECT_ID>.svc.id.goog[circleci-server/vm-service]"
+----
+
+. *Optionally, get JSON Key File*
++
+If you are using link:https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity[Workload Identities] for GKE, this step is not required.
++
+After running the following command, you should have a file named `circleci-server-vm-keyfile` in your local working directory. You will need this when you configure your server installation.
++
+[source,shell]
+----
+gcloud iam service-accounts keys create circleci-server-vm-keyfile --iam-account <YOUR_SERVICE_ACCOUNT_EMAIL>
+----
+
+. *Configure Server*
++
+Add the VM Service configuration to values.yaml. Details of the available configuration options can be found in the https://circleci.com/docs/2.0/server-3-operator-vm-service[VM Service] guide.
```

Comment:
This is showing as an external link? I might not have noticed this before in other links (plenty are styled as internal links), but this is styled as an external link.